hcl2template/addrs/plugin.go

```diff
@@ -13,21 +17,53 @@ import (
 
 // Plugin encapsulates a single plugin type.
 type Plugin struct {
-	Hostname  string
-	Namespace string
-	Type      string
+	Source string
 }
 
-func (p Plugin) RealRelativePath() string {
-	return p.Namespace + "/packer-plugin-" + p.Type
+// Parts returns the list of components of the source URL, starting with the
+// host, and ending with the name of the plugin (non-simplified).
+//
+// This will correspond more or less to the filesystem hierarchy where
+// the plugin is installed.
+func (p Plugin) Parts() []string {
+	var retParts []string
+
+	// This should not happen, we're supposed to check that the source isn't
+	// empty when creating the plugin, but let's be careful here and not
+	// misbehave.
+	if p.Source == "" {
+		log.Printf("[ERROR] - Plugin.Parts: the source was empty, this should not happen in normal usage, and is likely a Packer bug.")
```

Comment:
I see why logging is nice for debugging but it seem out of place given the use of this Method, at least to me. 

What about documenting the behavior in the comments.
```go
// Parts splits the plugin Source by URL path separator separating 
// it into hostname, namespace directories, and plugin name. If there 
// is no slash a slice containing the original is returned. On empty plugin Sources
// an empty slice is returned. It is up to the caller to check the length of the resulting slice.
func (p Plugin) Parts() []string {
```

Reply:
We can document the usage in the function itself yes, but that doesn't negate the fact that creating a `Plugin` instance with an empty string is a misuse of the structure, which we should probably be able to spot one way or another.
I'm not a fan of having `Parts` error/panic as if we have a bug this could be disruptive to users, but I think it's still valuable to at least warn through verbose logs that something went wrong.

Though I guess that implies users will need to look at the verbose logs to diagnose the problem (if perceived), so maybe this is unnecessary, unsure about this one.